### DISCUSSION TITLE
General question

### DISCUSSION THREAD

**Discussion by itsmohammadh** [Cyfrin/foundry-full-course-cu#855] [General]
hello everyone
A question for those who have completed the course, at the end of this course, according to dear Patrick training and personal research and effort, did you get the ability to write smart contracts and advanced test writing? And do you have the ability to read smart contracts of big projects? And did you get hired in a company or group (even in a junior or internship position)?
Given that you have no previous experience in programming!

**Comment by 0xusmanf** [comment]
> did you get the ability to write smart contracts and advanced test writing? And do you have the ability to read smart contracts of big projects?

Yes after completing this course I can write smart contracts and have the ability to write advanced tests like unit, integration, fuzz, and invariant testing.

I can also read smart contracts of big projects and I am working as a security researcher at [CodeHawks](https://www.codehawks.com/)

> And did you get hired in a company or group (even in a junior or internship position)?

Practically speaking, instead of thinking of getting a job or internship right after this course, you should think of this course as a starting point. You definitely have to dig deeper into solidity and build your portfolio before you start applying for internships and jobs.

> Given that you have no previous experience in programming!

I am speaking from the perspective of a person who had a little bit of a programming background before starting this course.

PS: If you are going to start this course then definitely try to complete all 15 lesson challenges and mint their NFTs, by doing so you will gain practical knowledge and will have a deeper understanding of the topics.

**Reply by itsmohammadh** [reply]
thank you usamn🙏❤️ goodluck
